### PR TITLE
Fibo 44 direct subclasses instances

### PR DESCRIPTION
**Pull request by patrycjamia**
Extract of Directed Subclasses and individuals for OWLClasses.

### CODE REVIEW

File: weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java [hunk 1]
```diff
@@ -63,14 +80,19 @@ public OwlDetails handleParticularClass(IRI iri, OWLOntology ontology) {
...
 
         OwlDetailsProperties<PropertyValue> axioms = handleAxioms(clazz, ontology);
 
         List<PropertyValue> subclasses = axioms.getProperties().getOrDefault(AxiomType.SUBCLASS_OF.getName(), new ArrayList<>(0));
         List<PropertyValue> taxElements = subclasses.stream().filter((pv) -> (pv.getType().equals(WeaselOwlType.TAXONOMY))).collect(Collectors.toList());
-        subclasses = subclasses.stream().filter((pv) -> (!pv.getType().equals(WeaselOwlType.TAXONOMY))).collect(Collectors.toList());
+        OwlDetailsProperties<PropertyValue> handleSubClassOf = handleParticularSubClassOf(ontology, clazz);
+        OwlDetailsProperties<PropertyValue> individuals = handleListIndividual(ontology, clazz);
 
+        //handleListIndividual(iri, ontology, clazz);
```
Comment (dasawanaka) [inline weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java]: @patrycjamia Delete unnecessary comments.

File: weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java [hunk 2]
```diff
@@ -398,4 +422,72 @@ public OwlDetails handleParticularObjectProperty(IRI iri, OWLOntology ontology)
...
+      // newResult.addAll(newResult);
+    }
+
+    return result;
+  }
+
+  private OwlDetailsProperties<PropertyValue> handleListIndividual(OWLOntology ontology, OWLClass clazz) {
+    OwlDetailsProperties<PropertyValue> result = new OwlDetailsProperties<PropertyValue>();
+    //Iterator<OWLNamedIndividual> iterator = ontology.individualsInSignature().iterator();
+    //List<OWLNamedIndividual> result = new LinkedList<>();
```
Comment (dasawanaka) [inline weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java]: @patrycjamia Delete unnecessary comments.

File: weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java [hunk 3]
```diff
@@ -398,4 +422,72 @@ public OwlDetails handleParticularObjectProperty(IRI iri, OWLOntology ontology)
...
+//      OWLNamedIndividual individual = iterator.next();
+//
+//      String fragment = iri.getFragment();
+//      LOGGER.debug(individual.getIRI().toString());
+//      result.add(individual);
+//      individual.getEntityType();
+//
+//    }
+//    return resultDetails;
+//  }
```
Comment (dasawanaka) [inline weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java]: @patrycjamia  Delete unnecessary comments. Lines 469 - 492

File: weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java [hunk 4]
```diff
@@ -398,4 +422,72 @@ public OwlDetails handleParticularObjectProperty(IRI iri, OWLOntology ontology)
...
+      String fragment = iri.getFragment();
+      LOGGER.debug(next.toString());
+      OWLClassExpression superClass = next.getSuperClass();
+//      result.addAll(result);
+      OwlDirectedSubClassesProperty r = new OwlDirectedSubClassesProperty();
+      r.setType(WeaselOwlType.DIRECT_SUBCLASSES);
+      r.setValue(new PairImpl(fragment, iri.toString()));
+      result.addProperty(WeaselOwlType.DIRECT_SUBCLASSES.name(), r);
+      // List<OwlGroupedDetails> newResult = new LinkedList<>();
+      // newResult.addAll(newResult);
```
Comment (dasawanaka) [inline weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java]: @patrycjamia Delete unnecessary comments. Lines: 429, 436, 441-442

**Review by trypuz — COMMENTED** [review]
Please use @author instead of "create by" (btw, it should be "created by").
Add comments to the code, please.

File: weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java [hunk 5]
```diff
@@ -398,4 +422,72 @@ public OwlDetails handleParticularObjectProperty(IRI iri, OWLOntology ontology)
...
+    }
+
+    return result;
+  }
+
+  private OwlDetailsProperties<PropertyValue> handleListIndividual(OWLOntology ontology, OWLClass clazz) {
```
Comment (dasawanaka) [inline weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java]: @patrycjamia You don't return the list, change to a more transparent name.

File: weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java [hunk 6]
```diff
@@ -398,4 +422,72 @@ public OwlDetails handleParticularObjectProperty(IRI iri, OWLOntology ontology)
...
+    OwlDetailsProperties<PropertyValue> result = new OwlDetailsProperties<PropertyValue>();
+    //Iterator<OWLNamedIndividual> iterator = ontology.individualsInSignature().iterator();
+    //List<OWLNamedIndividual> result = new LinkedList<>();
+    OWLReasonerFactory reasonerFactory = new StructuralReasonerFactory();
+    OWLReasoner reasoner = reasonerFactory.createNonBufferingReasoner(ontology);
+    NodeSet<OWLNamedIndividual> instances = reasoner.getInstances(clazz, true);
+    for (OWLNamedIndividual o : instances.entities().collect(Collectors.toSet())) {
+      String fragment = o.getIRI().getFragment();
+      LOGGER.debug(o.getIRI().toString());
```
Comment (dasawanaka) [inline weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/data/OwlDataHandler.java]: @patrycjamia Change the log to a more transparent for user, this information says nothing and may be misleading.

File: weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/WeaselOntologyManager.java [hunk 7]
```diff
@@ -185,19 +184,23 @@ public Collection getDetailsByIri(String iriString) {
       for (Map.Entry<String, List<PropertyValue>> entry : owlDetails.getProperties().entrySet()) {
         String propertyKey = entry.getKey();
         String groupName = null;
-
+        String groupSubClassOf = null;
         groupName = getGroupName(groups, propertyKey);
+        //groupSubClassOf = getDetailsByIri(groups, propertyKey);
```
Comment (dasawanaka) [inline weasel/src/main/java/org/edmcouncil/spec/fibo/weasel/ontology/WeaselOntologyManager.java]: @patrycjamia Delete unnecessary comments.

**Review by dasawanaka — CHANGES_REQUESTED** [review]
Please correct and submit again for approval.

**Pull request merged**